lib/streamlit/elements/time_widgets.py

```diff
@@ -233,6 +224,25 @@ def date_input(
             else:
                 max_value = date.today() + relativedelta.relativedelta(years=10)
 
+        if value:
```

Comment:
Is the idea that if an empty list/tuple is passed as value then we will create a date range picker with no default selected? I think in that case this validation isn't required and doesn't make sense for empty list/tuple.

Reply:
@lawilby Yes, if an empty list or tuple is passed as a value, then we do not specify default value in the widget.

I think there is `if value:` check especially for that case of empty list,  so that we confidently take value[0] and value[-1] without fear of getting an `Index out of the range` error at the next lines for `start_value` and `end_value`